### ISSUE TITLE
iOS: Undefined symbol _OPENSSL_armcap_P

### ISSUE THREAD

**Issue by misery** [openssl/openssl#22089]
We tried 3.2.0-alpha1 on our CI and the iOS build fails for arm64.

```
17:49:23 ${LDCMD:-cc} -arch arm64 -mios-version-min=11.0 -fno-common -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS.sdk -O3 -Os -fstack-protector-strong -fvisibility=hidden -L. -Wl,-search_paths_first  \
17:49:23 		-o apps/openssl \
17:49:23 		apps/lib/openssl-bin-cmp_mock_srv.o \
17:49:23 		apps/openssl-bin-asn1parse.o apps/openssl-bin-ca.o \
17:49:23 		apps/openssl-bin-ciphers.o apps/openssl-bin-cmp.o \
17:49:23 		apps/openssl-bin-cms.o apps/openssl-bin-crl.o \
17:49:23 		apps/openssl-bin-crl2pkcs7.o apps/openssl-bin-dgst.o \
17:49:23 		apps/openssl-bin-dhparam.o apps/openssl-bin-dsa.o \
17:49:23 		apps/openssl-bin-dsaparam.o apps/openssl-bin-ec.o \
17:49:23 		apps/openssl-bin-ecparam.o apps/openssl-bin-enc.o \
17:49:23 		apps/openssl-bin-errstr.o apps/openssl-bin-fipsinstall.o \
17:49:23 		apps/openssl-bin-gendsa.o apps/openssl-bin-genpkey.o \
17:49:23 		apps/openssl-bin-genrsa.o apps/openssl-bin-info.o \
17:49:23 		apps/openssl-bin-kdf.o apps/openssl-bin-list.o \
17:49:23 		apps/openssl-bin-mac.o apps/openssl-bin-nseq.o \
17:49:23 		apps/openssl-bin-ocsp.o apps/openssl-bin-openssl.o \
17:49:23 		apps/openssl-bin-passwd.o apps/openssl-bin-pkcs12.o \
17:49:23 		apps/openssl-bin-pkcs7.o apps/openssl-bin-pkcs8.o \
17:49:23 		apps/openssl-bin-pkey.o apps/openssl-bin-pkeyparam.o \
17:49:23 		apps/openssl-bin-pkeyutl.o apps/openssl-bin-prime.o \
17:49:23 		apps/openssl-bin-progs.o apps/openssl-bin-rand.o \
17:49:23 		apps/openssl-bin-rehash.o apps/openssl-bin-req.o \
17:49:23 		apps/openssl-bin-rsa.o apps/openssl-bin-s_client.o \
17:49:23 		apps/openssl-bin-s_server.o apps/openssl-bin-s_time.o \
17:49:23 		apps/openssl-bin-sess_id.o apps/openssl-bin-smime.o \
17:49:23 		apps/openssl-bin-speed.o apps/openssl-bin-spkac.o \
17:49:23 		apps/openssl-bin-storeutl.o apps/openssl-bin-verify.o \
17:49:23 		apps/openssl-bin-version.o apps/openssl-bin-x509.o \
17:49:23 		apps/libapps.a -lssl -lcrypto  
17:49:23 Undefined symbols for architecture arm64:
17:49:23   "_OPENSSL_armcap_P", referenced from:
17:49:23       _sha3_224_newctx in libcrypto.a(libdefault-lib-sha3_prov.o)
17:49:23       _sha3_256_newctx in libcrypto.a(libdefault-lib-sha3_prov.o)
17:49:23       _sha3_384_newctx in libcrypto.a(libdefault-lib-sha3_prov.o)
17:49:23       _sha3_512_newctx in libcrypto.a(libdefault-lib-sha3_prov.o)
17:49:23       _keccak_224_newctx in libcrypto.a(libdefault-lib-sha3_prov.o)
17:49:23       _keccak_256_newctx in libcrypto.a(libdefault-lib-sha3_prov.o)
17:49:23       _keccak_384_newctx in libcrypto.a(libdefault-lib-sha3_prov.o)
17:49:23       ...
17:49:23   "_SHA3_absorb_cext", referenced from:
17:49:23       _armsha3_sha3_absorb in libcrypto.a(libdefault-lib-sha3_prov.o)
17:49:23 ld: symbol(s) not found for architecture arm64
17:49:23 clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

```
17:49:04     perl ../openssl/Configure --prefix=/Users/governikus/workspace/default_Review_Libs_iOS/build/dist --libdir=lib no-ui-console no-filenames no-camellia no-bf no-aria no-seed no-poly1305 no-srp no-gost no-idea no-mdc2 no-rc2 no-rc4 no-rc5 no-srtp no-sm2 no-sm3 no-sm4 no-ct no-dgram no-cast no-chacha no-blake2 no-rmd160 no-scrypt no-siphash no-whirlpool no-md4 no-des no-ec2m no-tls1 no-tls1-method no-tls1_1 no-tls1_1-method no-tls1_3 no-ssl3 no-ssl3-method no-dtls no-dtls1-method no-dtls1_2-method no-deprecated no-engine no-async no-dso no-comp no-ts no-makedepend no-tests no-legacy shared no-asm ios64-cross -Os -fstack-protector-strong -fvisibility=hidden
17:49:04 
17:49:04 Perl information:
17:49:04 
17:49:04     perl
17:49:04     5.30.3 for darwin-thread-multi-2level
17:49:04 
17:49:04 Enabled features:
17:49:04 
17:49:04     apps
17:49:04     autoalginit
17:49:04     autoerrinit
17:49:04     autoload-config
17:49:04     bulk
17:49:04     cached-fetch
17:49:04     cmac
17:49:04     cmp
17:49:04     cms
17:49:04     default-thread-pool
17:49:04     dh
17:49:04     docs
17:49:04     dsa
17:49:04     ec
17:49:04     ecdh
17:49:04     ecdsa
17:49:04     ecx
17:49:04     err
17:49:04     http
17:49:04     multiblock
17:49:04     nextprotoneg
17:49:04     ocb
17:49:04     ocsp
17:49:04     pic
17:49:04     pinshared
17:49:04     posix-io
17:49:04     psk
17:49:04     rdrand
17:49:04     rfc3779
17:49:04     secure-memory
17:49:04     siv
17:49:04     sm2-precomp
17:49:04     sock
17:49:04     sse2
17:49:04     ssl
17:49:04     ssl-trace
17:49:04     static-engine
17:49:04     stdio
17:49:04     thread-pool
17:49:04     threads
17:49:04     tls
17:49:04     tls1_2
17:49:04     tls1_2-method
17:49:04 
17:49:04 Disabled features:
17:49:04 
17:49:04     acvp-tests          [cascade]                OPENSSL_NO_ACVP_TESTS
17:49:04     afalgeng            [cascade]                OPENSSL_NO_AFALGENG
17:49:04     argon2              [cascade]                OPENSSL_NO_ARGON2
17:49:04     aria                [option]                 OPENSSL_NO_ARIA (skip crypto/aria)
17:49:04     asan                [default]                OPENSSL_NO_ASAN
17:49:04     asm                 [option]                 OPENSSL_NO_ASM
17:49:04     async               [config]                 OPENSSL_NO_ASYNC
17:49:04     bf                  [option]                 OPENSSL_NO_BF (skip crypto/bf)
17:49:04     blake2              [option]                 OPENSSL_NO_BLAKE2
17:49:04     brotli              [default]                OPENSSL_NO_BROTLI
17:49:04     brotli-dynamic      [default]                OPENSSL_NO_BROTLI_DYNAMIC
17:49:04     buildtest-c++       [default]                
17:49:04     camellia            [option]                 OPENSSL_NO_CAMELLIA (skip crypto/camellia)
17:49:04     capieng             [cascade]                OPENSSL_NO_CAPIENG
17:49:04     winstore            [not-windows]            OPENSSL_NO_WINSTORE
17:49:04     cast                [option]                 OPENSSL_NO_CAST (skip crypto/cast)
17:49:04     chacha              [option]                 OPENSSL_NO_CHACHA (skip crypto/chacha)
17:49:04     comp                [option]                 OPENSSL_NO_COMP (skip crypto/comp)
17:49:04     crypto-mdebug       [default]                OPENSSL_NO_CRYPTO_MDEBUG
17:49:04     ct                  [option]                 OPENSSL_NO_CT (skip crypto/ct)
17:49:04     deprecated          [option]                 OPENSSL_NO_DEPRECATED
17:49:04     des                 [option]                 OPENSSL_NO_DES (skip crypto/des)
17:49:04     devcryptoeng        [default]                OPENSSL_NO_DEVCRYPTOENG
17:49:04     dgram               [option]                 OPENSSL_NO_DGRAM
17:49:04     dso                 [option]                 OPENSSL_NO_DSO
17:49:04     dtls                [option(dtls)]           OPENSSL_NO_DTLS
17:49:04     dynamic-engine      [cascade]                
17:49:04     ec2m                [option]                 OPENSSL_NO_EC2M
17:49:04     ec_nistp_64_gcc_128 [default]                OPENSSL_NO_EC_NISTP_64_GCC_128
17:49:04     egd                 [default]                OPENSSL_NO_EGD
17:49:04     engine              [option]                 OPENSSL_NO_ENGINE (skip crypto/engine)
17:49:04     external-tests      [default]                OPENSSL_NO_EXTERNAL_TESTS
17:49:04     filenames           [option]                 OPENSSL_NO_FILENAMES
17:49:04     fips                [default]                
17:49:04     fips-securitychecks [cascade]                OPENSSL_NO_FIPS_SECURITYCHECKS
17:49:04     fuzz-afl            [default]                OPENSSL_NO_FUZZ_AFL
17:49:04     fuzz-libfuzzer      [default]                OPENSSL_NO_FUZZ_LIBFUZZER
17:49:04     gost                [option]                 OPENSSL_NO_GOST
17:49:04     idea                [option]                 OPENSSL_NO_IDEA (skip crypto/idea)
17:49:04     ktls                [default]                OPENSSL_NO_KTLS
17:49:04     legacy              [option]                 
17:49:04     loadereng           [cascade]                OPENSSL_NO_LOADERENG
17:49:04     makedepend          [option]                 
17:49:04     md2                 [default]                OPENSSL_NO_MD2 (skip crypto/md2)
17:49:04     md4                 [option]                 OPENSSL_NO_MD4 (skip crypto/md4)
17:49:04     mdc2                [option]                 OPENSSL_NO_MDC2 (skip crypto/mdc2)
17:49:04     module              [cascade]                
17:49:04     msan                [default]                OPENSSL_NO_MSAN
17:49:04     padlockeng          [cascade]                OPENSSL_NO_PADLOCKENG
17:49:04     poly1305            [option]                 OPENSSL_NO_POLY1305 (skip crypto/poly1305)
17:49:04     quic                [cascade]                OPENSSL_NO_QUIC
17:49:04     rc2                 [option]                 OPENSSL_NO_RC2 (skip crypto/rc2)
17:49:04     rc4                 [option]                 OPENSSL_NO_RC4 (skip crypto/rc4)
17:49:04     rc5                 [option]                 OPENSSL_NO_RC5 (skip crypto/rc5)
17:49:04     rmd160              [option]                 OPENSSL_NO_RMD160 (skip crypto/ripemd)
17:49:04     scrypt              [option]                 OPENSSL_NO_SCRYPT
17:49:04     sctp                [default]                OPENSSL_NO_SCTP
17:49:04     seed                [option]                 OPENSSL_NO_SEED (skip crypto/seed)
17:49:04     shared              [config]                 
17:49:04     siphash             [option]                 OPENSSL_NO_SIPHASH (skip crypto/siphash)
17:49:04     sm2                 [option]                 OPENSSL_NO_SM2 (skip crypto/sm2)
17:49:04     sm3                 [option]                 OPENSSL_NO_SM3 (skip crypto/sm3)
17:49:04     sm4                 [option]                 OPENSSL_NO_SM4 (skip crypto/sm4)
17:49:04     srp                 [option]                 OPENSSL_NO_SRP (skip crypto/srp)
17:49:04     srtp                [option]                 OPENSSL_NO_SRTP
17:49:04     tests               [option]                 OPENSSL_NO_TESTS
17:49:04     tfo                 [default]                OPENSSL_NO_TFO
17:49:04     trace               [default]                OPENSSL_NO_TRACE
17:49:04     ts                  [option]                 OPENSSL_NO_TS (skip crypto/ts)
17:49:04     ubsan               [default]                OPENSSL_NO_UBSAN
17:49:04     ui-console          [option]                 OPENSSL_NO_UI_CONSOLE
17:49:04     unit-test           [default]                OPENSSL_NO_UNIT_TEST
17:49:04     uplink              [no uplink_arch]         OPENSSL_NO_UPLINK
17:49:04     weak-ssl-ciphers    [default]                OPENSSL_NO_WEAK_SSL_CIPHERS
17:49:04     whirlpool           [option]                 OPENSSL_NO_WHIRLPOOL (skip crypto/whrlpool)
17:49:04     zlib                [default]                OPENSSL_NO_ZLIB
17:49:04     zlib-dynamic        [default]                OPENSSL_NO_ZLIB_DYNAMIC
17:49:04     zstd                [default]                OPENSSL_NO_ZSTD
17:49:04     zstd-dynamic        [default]                OPENSSL_NO_ZSTD_DYNAMIC
17:49:04     ssl3                [option(ssl3-method)]    OPENSSL_NO_SSL3
17:49:04     ssl3-method         [option]                 OPENSSL_NO_SSL3_METHOD
17:49:04     tls1                [option(tls1-method)]    OPENSSL_NO_TLS1
17:49:04     tls1-method         [option]                 OPENSSL_NO_TLS1_METHOD
17:49:04     tls1_1              [option(tls1_1-method)]  OPENSSL_NO_TLS1_1
17:49:04     tls1_1-method       [option]                 OPENSSL_NO_TLS1_1_METHOD
17:49:04     tls1_3              [option]                 OPENSSL_NO_TLS1_3
17:49:04     dtls1               [option(dtls1-method)]   OPENSSL_NO_DTLS1
17:49:04     dtls1-method        [option]                 OPENSSL_NO_DTLS1_METHOD
17:49:04     dtls1_2             [option(dtls1_2-method)] OPENSSL_NO_DTLS1_2
17:49:04     dtls1_2-method      [option]                 OPENSSL_NO_DTLS1_2_METHOD
17:49:04 
17:49:04 Config target attributes:
17:49:04 
17:49:04     AR => "ar",
17:49:04     ARFLAGS => "qc",
17:49:04     CC => "cc",
17:49:04     CFLAGS => "-O3",
17:49:04     HASHBANGPERL => "/usr/bin/env perl",
17:49:04     RANLIB => "ranlib -c",
17:49:04     RC => "windres",
17:49:04     asm_arch => "aarch64",
17:49:04     bn_ops => "SIXTY_FOUR_BIT_LONG RC4_CHAR",
17:49:04     build_file => "Makefile",
17:49:04     build_scheme => [ "unified", "unix" ],
17:49:04     cflags => "-arch arm64 -mios-version-min=11.0 -fno-common -isysroot \$(CROSS_TOP)/SDKs/\$(CROSS_SDK)",
17:49:04     cppflags => "-D_REENTRANT",
17:49:04     defines => [ "OPENSSL_BUILDING_OPENSSL" ],
17:49:04     disable => [ "shared", "async" ],
17:49:04     dso_scheme => "dlfcn",
17:49:04     enable => [  ],
17:49:04     ex_libs => "",
17:49:04     includes => [  ],
17:49:04     lflags => "-Wl,-search_paths_first",
17:49:04     lib_cflags => "",
17:49:04     lib_cppflags => "",
17:49:04     lib_defines => [  ],
17:49:04     module_cflags => "-fPIC",
17:49:04     module_cxxflags => undef,
17:49:04     module_ldflags => "-bundle",
17:49:04     perl_platform => "Unix",
17:49:04     perlasm_scheme => "ios64",
17:49:04     shared_cflag => "-fPIC",
17:49:04     shared_defines => [  ],
17:49:04     shared_extension => ".\$(SHLIB_VERSION_NUMBER).dylib",
17:49:04     shared_ldflag => "-dynamiclib -current_version \$(SHLIB_VERSION_NUMBER) -compatibility_version \$(SHLIB_VERSION_NUMBER)",
17:49:04     shared_rcflag => "",
17:49:04     shared_sonameflag => "-install_name \$(libdir)/",
17:49:04     shared_target => "darwin-shared",
17:49:04     sys_id => "iOS",
17:49:04     thread_defines => [  ],
17:49:04     thread_scheme => "pthreads",
17:49:04     unistd => "<unistd.h>",
17:49:04 
17:49:04 Recorded environment:
17:49:04 
17:49:04     AR = 
17:49:04     BUILDFILE = 
17:49:04     CC = 
17:49:04     CFLAGS = 
17:49:04     CPPFLAGS = 
17:49:04     CROSS_COMPILE = 
17:49:04     CXX = 
17:49:04     CXXFLAGS = 
17:49:04     HASHBANGPERL = 
17:49:04     LDFLAGS = 
17:49:04     LDLIBS = 
17:49:04     OPENSSL_LOCAL_CONFIG_DIR = 
17:49:04     PERL = 
17:49:04     RANLIB = 
17:49:04     RC = 
17:49:04     RCFLAGS = 
17:49:04     WINDRES = 
17:49:04     __CNF_CFLAGS = 
17:49:04     __CNF_CPPDEFINES = 
17:49:04     __CNF_CPPFLAGS = 
17:49:04     __CNF_CPPINCLUDES = 
17:49:04     __CNF_CXXFLAGS = 
17:49:04     __CNF_LDFLAGS = 
17:49:04     __CNF_LDLIBS = 
17:49:04 
17:49:04 Makevars:
17:49:04 
17:49:04     AR              = ar
17:49:04     ARFLAGS         = qc
17:49:04     ASFLAGS         = 
17:49:04     CC              = cc
17:49:04     CFLAGS          = -O3 -Os -fstack-protector-strong -fvisibility=hidden
17:49:04     CPPDEFINES      = 
17:49:04     CPPFLAGS        = 
17:49:04     CPPINCLUDES     = 
17:49:04     CXXFLAGS        = -Os -fstack-protector-strong -fvisibility=hidden
17:49:04     HASHBANGPERL    = /usr/bin/env perl
17:49:04     LDFLAGS         = 
17:49:04     LDLIBS          = 
17:49:04     PERL            = perl
17:49:04     RANLIB          = ranlib -c
17:49:04     RC              = windres
17:49:04     RCFLAGS         = 
17:49:04 
17:49:04 NOTE: These variables only represent the configuration view.  The build file
17:49:04 template may have processed these variables further, please have a look at the
17:49:04 build file for more exact data:
17:49:04     Makefile
17:49:04 
17:49:04 build file:
17:49:04 
17:49:04     Makefile
17:49:04 
17:49:04 build file templates:
17:49:04 
17:49:04     ../openssl/Configurations/common0.tmpl
17:49:04     ../openssl/Configurations/unix-Makefile.tmpl
```

**Comment by t8m** [comment]
Please try the fix in #22090

**Comment by misery** [comment]
Thanks, it works.
